src/nameTable.java

```diff
@@ -3,21 +3,38 @@
 import src.parseTree.tokens.id;
 
 import java.util.HashMap;
+import java.util.Map;
 
 public class nameTable {
+    private static nameTable instance;
+    private static String filePath;
 
-    private HashMap<String, typeIdx> typeMap;
-    private HashMap<String, Integer> intMap;
-    private HashMap<String, Double> doubleMap;
-    private HashMap<String, String> strMap;
+    private Map<String, typeIdx> typeMap;
+    private Map<String, Integer> intMap;
+    private Map<String, Double> doubleMap;
+    private Map<String, String> strMap;
 
-    public nameTable() {
+
+    public static void init_nameTable(String filepath) {
```

Comment:
We need to call this somewhere to init the first time. Probably would be fine to call as the the first step in `#tokenizer.tokenize`  since tokenization is always going to be our first step of execution.

Reply:
Probably better to do it at start of parsing. the tokenizer doesn't make use of it, and if it fails in the tokenizer stage we may as well save ourselves the effort

Reply:
And this will end up being done on the parser path. Execution can assume it's done, because it's part of the parser's job